crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -53,84 +52,752 @@ pub struct CallContractCommand {
+	/// Recommended for testing and local development only.
+	#[clap(name = "dev", long, short, default_value = "false")]
+	dev_mode: bool,
 }
-
 impl CallContractCommand {
 	/// Executes the command.
-	pub(crate) async fn execute(self) -> anyhow::Result<()> {
-		clear_screen()?;
-		intro(format!("{}: Calling a contract", style(" Pop CLI ").black().on_magenta()))?;
-		set_theme(Theme);
-
-		let call_exec = set_up_call(CallOpts {
-			path: self.path.clone(),
-			contract: self.contract.clone(),
-			message: self.message.clone(),
-			args: self.args.clone(),
-			value: self.value.clone(),
-			gas_limit: self.gas_limit,
-			proof_size: self.proof_size,
-			url: self.url.clone(),
-			suri: self.suri.clone(),
-			execute: self.execute,
-		})
-		.await?;
+	pub(crate) async fn execute(self) -> Result<()> {
+		let call_config: CallContractCommand = match self.set_up_call_config(&mut cli::Cli).await {
+			Ok(call_config) => call_config,
+			Err(e) => {
+				display_message(&e.to_string(), false, &mut cli::Cli)?;
+				return Ok(());
+			},
+		};
+		match execute_call(call_config, self.contract.is_none(), &mut cli::Cli).await {
+			Ok(_) => Ok(()),
+			Err(e) => {
+				display_message(&e.to_string(), false, &mut cli::Cli)?;
+				Ok(())
+			},
+		}
+	}
 
+	fn display(&self) -> String {
+		let mut full_message = "pop call contract".to_string();
+		if let Some(path) = &self.path {
+			full_message.push_str(&format!(" --path {}", path.display()));
+		}
+		if let Some(contract) = &self.contract {
+			full_message.push_str(&format!(" --contract {}", contract));
+		}
+		if let Some(message) = &self.message {
+			full_message.push_str(&format!(" --message {}", message));
+		}
+		if !self.args.is_empty() {
+			full_message.push_str(&format!(" --args {}", self.args.join(" ")));
+		}
+		if self.value != "0" {
+			full_message.push_str(&format!(" --value {}", self.value));
+		}
+		if let Some(gas_limit) = self.gas_limit {
+			full_message.push_str(&format!(" --gas {}", gas_limit));
+		}
+		if let Some(proof_size) = self.proof_size {
+			full_message.push_str(&format!(" --proof_size {}", proof_size));
+		}
+		full_message.push_str(&format!(" --url {} --suri {}", self.url, self.suri));
+		if self.execute {
+			full_message.push_str(" --execute");
+		}
 		if self.dry_run {
+			full_message.push_str(" --dry_run");
+		}
+		full_message
+	}
+
+	/// Set up the config call.
+	async fn set_up_call_config(
+		&self,
+		cli: &mut impl cli::traits::Cli,
+	) -> anyhow::Result<CallContractCommand> {
+		cli.intro("Call a contract")?;
+		let call_config = if self.contract.is_none() {
+			match guide_user_to_call_contract(None, None, None, self.dev_mode, cli).await {
+				Ok(config) => config,
+				Err(e) => {
+					return Err(anyhow!(format!("{}", e.to_string())));
+				},
+			}
+		} else {
+			self.clone()
+		};
+		Ok(call_config)
+	}
+}
+
+/// Guide the user to call the contract.
+async fn guide_user_to_call_contract(
```

Comment:
For the command.field.is_none() in case of `url`, `suri` and `value` I check is not the default value to prompt for it again, because we have a default_value for using the command line